### ISSUE TITLE
Memory leak when evaluating model on CPU

### ISSUE THREAD

**Issue by williamdias** [pytorch/pytorch#40973]
## 🐛 Bug

Pytorch seems to be leaking memory when evaluating a model on cpu.

![Screen Shot 2020-07-03 at 17 28 51](https://user-images.githubusercontent.com/686915/86496455-e8916780-bd53-11ea-8c01-cbbab95bce5c.png)

## To Reproduce

```python
import numpy as np
import cv2
import os
import torch
import torchvision.models as models

current_path = os.path.dirname(os.path.abspath(__file__))
testset_folder = os.path.join(current_path, '../datasets/widerface/validation/images')
testset_list = os.path.join(current_path, '../datasets/widerface/validation/wider_val.txt')

with open(testset_list, 'r') as fr:
    test_dataset = fr.read().split()

torch.set_grad_enabled(False)
model = models.resnet50()
model.eval()

device = torch.device('cpu')
model = model.to(device)

for i, filename in enumerate(test_dataset):
    print('{}/{}'.format(i, len(test_dataset)))
    img_path = os.path.join(testset_folder, filename)
    img_raw = cv2.imread(img_path, cv2.IMREAD_COLOR)
    img = np.float32(img_raw)
    img = img.transpose(2, 0, 1)
    img = torch.from_numpy(img).unsqueeze(0)
    img = img.to(device)
    model(img)
```

## Expected behavior

I expected memory consumption would reach a plateau after some time and not keep growing indefinitely. 

## Environment

PyTorch version: 1.5.1+cu101
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.4 LTS
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 430.50
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.5.1+cu101
[pip3] torchvision==0.6.1+cu101
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h6bb024c_0  
[conda] mkl                       2020.1                      217  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.1.0            py37h23d657b_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] numpy                     1.18.1           py37h4f9e942_0  
[conda] numpy-base                1.18.1           py37hde5b4d6_1  
[conda] torch                     1.5.1+cu101              pypi_0    pypi
[conda] torchvision               0.6.1+cu101              pypi_0    pypi

## Additional context

I am running the provided code inside a docker container.

cc @ezyang @gchanan @zou3519

**Comment by ngimel** [comment]
Setting high priority because of memory leak. cc @ilia-cher in case it is omp-related.

**Comment by williamdias** [comment]
Apparently the nightly build fixes the problem. I've performed the same test with the preview version and the memory consumption seems to be stable.

**Comment by ilia-cher** [comment]
do you see any difference if you use `torch.set_num_threads(1)`?

**Comment by williamdias** [comment]
@ilia-cher

**Comment by mattip** [comment]
Maybe connected to gh-32284 which was fixed in gh-36479, merged as commit cc3fc786b7 and released in 1.6.0rc1?

**Comment by ilia-cher** [comment]
closing for now, if you notice this on a fresh build, please reopen
